### PR TITLE
feat(nba): human-readable player and team columns for model outputs

### PR DESCRIPTION
**Pull request by saiemgilani**
`nba_player_impact` shipped as a wall of `player_id`s. Validating it, the WAR leaderboard read `1628983` and I had to recognise the number to know it was Shai Gilgeous-Alexander — every consumer would have to rebuild the same crosswalk.

## What

`nba_player_identity()` derives `player_name`, `team_id`, `team_abbreviation`, `team_name` and `teams` from a season's own game logs.

Two consequences follow from that source choice:

- **Season-accurate.** A player *directory* reports whoever the player suits up for **today**, which would mislabel every historical season. Taking it from that season's logs means 1996-97 correctly reads *Michael Jordan / CHI / Chicago Bulls*.
- **Trades stay visible.** `team_*` is the club he logged the most minutes for; `teams` lists each stop in descending minutes (`"MIA,PHI"`) rather than collapsing to one and losing the other. Verified against Kyle Lowry and P.J. Tucker 2023-24.

Ties in the minutes sort break on `team_id` — `group_by` output order alone is not deterministic, so without that a rebuild would not be byte-identical.

Empty or missing-column input returns the typed zero-row frame, so callers can join unconditionally.

## Published result

30 seasons rebuilt and republished; verified through the live loader:

```
1997: Michael Jordan          CHI  Chicago Bulls          32.3 WAR
2025: Shai Gilgeous-Alexander OKC  Oklahoma City Thunder  21.9 WAR
```

Name coverage is complete except **4 rows of 20,660** (1997, 2018) that appear in possession lineups but have no box-log rows at all (`min`/`gp` are null too). That is an upstream gap, and they are left null rather than dropped — losing a player who has a real RAPM would be worse than an honest null.

## Note

The documented schema is edited in `tools/codegen/schemas/loader_schemas.yaml`. `nba_loaders.py` is generated and marked `DO NOT EDIT` — I edited it first and the next codegen run silently reverted it.

`704 passed`; ruff + mypy clean. Producer side: sportsdataverse/hoopR-nba-stats-data.

## Summary by Sourcery

Add a helper to derive season-accurate player and team identity metadata from NBA box logs and expose it in RAPM model outputs.

New Features:
- Introduce nba_player_identity helper to compute human-readable player and team identity columns from per-game box logs.
- Expose PLAYER_IDENTITY_SCHEMA and nba_player_identity through the public NBA module and parsed namespace.
- Augment nba_player_impact loader outputs with player_name and team-level identity columns for easier downstream use.

Enhancements:
- Document nba_player_identity in the NBA additional reference and update loader documentation to describe the enriched nba_player_impact schema.
- Increase the documented count of additional NBA helper functions to reflect the new API.

Tests:
- Add tests covering traded-player handling, ordering, and empty/malformed-input behavior for nba_player_identity.

## Summary by CodeRabbit

- **New Features**
  - Added NBA player identity enrichment that returns season-accurate player identity with player name and team details.
  - Correctly handles traded players by selecting a “primary” team by minutes and listing all teams played for in minutes order.
  - Exposed via the main NBA API and the legacy deprecated namespace.

- **Documentation**
  - Documented the new identity function and its input requirements and behavior for empty/missing-column inputs.
  - Updated `load_nba_player_impact` return schema to include player and team fields.
  - Updated NBA API function summary count.

- **Tests**
  - Added and strengthened identity and loader schema validation tests (including ordering and typed empty-frame behavior).

### CODE REVIEW

File: tests/nba/test_nba_box_logs.py [hunk 1]
```diff
@@ -172,3 +172,52 @@ def test_full_game_player_gets_the_teams_possessions():
...
+
+    from sportsdataverse.nba.nba_box_logs import (
+        PLAYER_IDENTITY_SCHEMA,
+        nba_player_identity,
+    )
+
+    for bad in (pl.DataFrame(), pl.DataFrame({"player_id": [1]})):
+        out = nba_player_identity(bad)
+        assert out.height == 0
+        assert out.columns == list(PLAYER_IDENTITY_SCHEMA)
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tests/nba/test_nba_box_logs.py]: Valid — fixed in `8123b096`. Columns alone didn't protect the contract: a Null-typed `player_id` would have passed while breaking a downstream join. Now asserts `dict(out.schema) == PLAYER_IDENTITY_SCHEMA`, and the case list grew the two that matter — a frame missing `min` and one missing `team_name`.

File: tools/codegen/schemas/loader_schemas.yaml [hunk 2]
```diff
@@ -4458,6 +4458,16 @@ load_nba_player_boxscore:
...
+- name: player_name
+  type: Utf8
+- name: team_id
+  type: Int64
+- name: team_abbreviation
+  type: Utf8
+- name: team_name
+  type: Utf8
+- name: teams
+  type: Utf8
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tools/codegen/schemas/loader_schemas.yaml:4470]: Valid — fixed in `8123b096`. `_IMPACT_SCHEMA` now carries the six identity columns, so the round-trip test actually validates the contract this PR changes rather than passing around it.

Its fixture also needed a fix to accept them: it built every column with `pl.Series([0], dtype=t)`, which cannot construct a `Utf8` series (`TypeError: 'int' object is not an instance of 'str'`). Each column is now seeded with a value of its own dtype.

File: sportsdataverse/nba/nba_box_logs.py [hunk 3]
```diff
@@ -128,3 +128,83 @@ def nba_box_logs(
...
+    per_team = (
+        player_logs.with_columns(minutes.fill_null(0.0).cast(pl.Float64).alias("_min"))
+        .group_by(["player_id", "team_id", "team_abbreviation", "team_name"])
+        .agg(pl.col("_min").sum().alias("_team_min"), pl.col("player_name").last().alias("player_name"))
+        # Ties would otherwise resolve by group_by's nondeterministic output order,
+        # so break them on team_id to keep a rebuild byte-identical.
+        .sort(["player_id", "_team_min", "team_id"], descending=[False, True, False])
+    )
+    primary = per_team.group_by("player_id").first()
+    teams = per_team.group_by("player_id").agg(pl.col("team_abbreviation").str.join(",").alias("teams"))
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/nba/nba_box_logs.py]: Valid, and the more serious of the two correctness findings — fixed in `8123b096`. You're right that the pre-sort wasn't load-bearing in the way I assumed: **both** the `teams` string *and* the primary-team pick via `.first()` were resting on `group_by` preserving that order, which polars only guarantees with `maintain_order`.

I probed it with 300 players (high-minutes team deliberately given the *lower* `team_id`) and it picked correctly 300/300 — which is exactly what makes this dangerous: undocumented behaviour that holds until data size or threading changes it.

Rather than add `maintain_order`, both now sort **inside** the aggregation (`sort_by(["_team_min","team_id"], descending=[True,False]).first()`), which is correct regardless of group ordering. Pinned by `test_player_identity_primary_team_does_not_depend_on_group_order` (200 players, id-ordered fallback would fail it).

File: sportsdataverse/nba/nba_box_logs.py [hunk 4]
```diff
@@ -128,3 +128,83 @@ def nba_box_logs(
...
+        .. _nba_api: https://github.com/swar/nba_api
+        .. _hoopR: https://hoopR.sportsdataverse.org
+    """
+    need = {"player_id", "player_name", "team_id", "team_abbreviation", "team_name"}
+    if player_logs.is_empty() or not need.issubset(set(player_logs.columns)):
+        return pl.DataFrame(schema=PLAYER_IDENTITY_SCHEMA)
+
+    minutes = pl.col("min") if "min" in player_logs.columns else pl.lit(0.0)
+    per_team = (
+        player_logs.with_columns(minutes.fill_null(0.0).cast(pl.Float64).alias("_min"))
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/nba/nba_box_logs.py]: Valid — fixed in `8123b096`, and this was a genuine silent-degradation path. With `min` absent every team aggregated to zero minutes, so "primary team" quietly became whichever `team_id` sorted first: an answer shaped like a right one. `min` is now required and a frame lacking it returns the typed zero-row frame.

Worth noting: **my own new regression test caught that this guard hadn't actually landed.** The edit adding `min` to the required set failed silently, so the docstring promised a check the code never performed — the test failed with `ColumnNotFoundError` and exposed it.

File: sportsdataverse/nba/nba_box_logs.py [hunk 5]
```diff
@@ -128,3 +128,83 @@ def nba_box_logs(
...
+            from sportsdataverse.nba import nba_box_logs, nba_player_identity
+            logs = nba_box_logs("2023-24")
+            named = ratings.join(nba_player_identity(logs["player"]), on="player_id", how="left")
+
+        See Also:
+            * `nba_api`_ -- reference Python client for stats.nba.com
+            * `hoopR`_ -- R companion package for NBA/MBB data
+
+        .. _nba_api: https://github.com/swar/nba_api
+        .. _hoopR: https://hoopR.sportsdataverse.org
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/nba/nba_box_logs.py:200]: Valid — fixed in `8123b096`. The example referenced an undefined `ratings` and would `NameError` on copy. It now builds both frames inline (a real SGA row) so it runs as-is, and a `Raises:` section documents that a malformed frame yields the typed zero-row frame rather than raising — which is the contract callers depend on to join unconditionally.
